compiler/rustc_hir/src/hir_id.rs

```diff
@@ -1,6 +1,43 @@
-use crate::def_id::{LocalDefId, CRATE_DEF_ID};
+use crate::def_id::{DefId, LocalDefId, CRATE_DEF_ID};
+use rustc_data_structures::stable_hasher::{HashStable, StableHasher, ToStableHashKey};
+use rustc_span::{def_id::DefPathHash, HashStableContext};
 use std::fmt;
 
+#[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
+#[derive(Encodable, Decodable)]
+pub struct OwnerId {
+    pub def_id: LocalDefId,
+}
+
+impl OwnerId {
+    #[inline]
+    pub fn to_def_id(self) -> DefId {
+        self.def_id.to_def_id()
+    }
+}
+
+impl From<LocalDefId> for OwnerId {
+    fn from(def_id: LocalDefId) -> Self {
```

Comment:
I'm not really fond of this `into` impl.  The point of `OwnerId` is to make a strong typedef around `LocalDefId` for those that appear as `owner` in a `HirId`.  Is there a way to use a more descriptive name?
OTOH, I'd welcome a `impl From<OwnerId> for HirId`.